### DISCUSSION TITLE
Vercel sending uncompressed data to Cloudflare

### DISCUSSION THREAD

**Discussion by andreivreja** [vercel/community#162] [Help] [answered]
After around a month of debugging my bandwidth bills (well, usage), figured out why I see inflated bandwidth usage numbers on Vercel (and so does everyone proxying them through Cloudflare).

All traffic that goes from Vercel to Cloudflare is sent uncompressed (compression is disabled on purpose). This is done by simply sending a raw response to any request that contains a `cf-ray` header combined with an `accept-encoding: gzip` (exactly what you'd get on an origin, from Cloudflare).

I'd like to believe there's a technical reason behind this, but it honestly just feels like something done to offset the cost savings provided by Cloudflare, by offsetting the lost usage with uncompressed data. That's just how I feel though, I'd love to hear your thoughts, and preferably some official answer/stance on this. 

Lets remember, $55/100GB is the pricing.

**Comment by leerob** [comment, marked as answer]
Thank you for providing this information to help us further investigate the issue!

Our infrastructure team has used the information you provided to determine an improvement (working with the Cloudflare team). We have made a change to improve how we handle compression between Vercel and Cloudflare.

Our infrastructure now has more granular control over compression, and thus can decrease bandwidth between Vercel and Cloudflare.

Starting today (Nov 24th, 2021), you should start to see lower bandwidth on your Usage page from these changes.

---

https://github.com/vercel/community/discussions/163#discussioncomment-1696358

**Reply by arunesh90** [reply]
Is it possible to get more information on what exactly caused Cloudflare traffic to get handled differently?

**Reply by samcx** [reply]
To clarify, this was indeed related to a past, technical decision made with regards to fully supporting Cloudflare proxy on top of Vercel (related to video playback support browsers when media is gzip compressed, of all things).

Without getting too much into the technical details, we were, until now, sending uncompressed traffic to Cloudflare, to fully support Cloudflare proxy on top of Vercel. 

Thanks again to @andreivreja for bringing awareness to this issue!

**Reply by andreivreja** [reply]
That’s some good news, thanks for looking into it! However, there’s still some confusion around it (at least for me).

> Starting today (Nov 24th, 2021), you should start to see lower bandwidth on your Usage page from these changes.

Does this mean that Vercel now treats Cloudflare like any other gzip compatible client, all the time? Emphasis on “all the time”. If there are still scenarios when this is not the case, can we know them?

It mostly comes down to transparency and feeling safe that we are being charged fairly. Most of your customer would reasonably think that Cloudflare would be treated equally. Most of those customers also, most probably, never noticed their usage was much higher than what it would’ve been if sending compression data.

Anyone would reasonably assume that the bandwidth usage wouldn’t be higher towards Cloudflare, resulting in a “hidden overcharge”. Will customers affected by this be informed privately (or via a public announcement)? Will their past bills (not mine, Vercel team already waived my first bill citing “unexpected usage growth”) be rectified?

**Reply by mcsdevv** [reply]
There are no other exceptions remaining. Vercel now treats Cloudflare like any other gzip compatible client in all cases.

We will be reaching out to all customers who, as a result of being impacted by this, have incurred additional costs due to resource usage.

**Reply by andreivreja** [reply]
That’s amazing to hear, thanks for addressing this!

**Reply by mcsdevv** [reply]
Thank **you** @andreivreja for raising this and your patience in allowing us to resolve, it is appreciated.
